### DISCUSSION TITLE
Notify if human voice heard

### DISCUSSION THREAD

**Discussion by London-Lass** [mcguirepr89/BirdNET-Pi#1086] [Ideas]
Audio feed to birdnet is from a wildlife livestream. 

For privacy purposes, I would like the ability to detect human voices and mute the stream. I realised looking at the config that birdnet has the capability to detect human voices, since it offers to discard if voice is heard.

Is there any trigger that could be made available via Apprise so that I could also post to a URL/Endpoint and thereby I could use that as an action to mute the audio for a period of time?

I guess if Human is considered a detectable species this might be the equivalent of species specific notification? Although I only need this for human voice.

**Comment by tphakala** [comment]
I have same use case and I will most likely implement this to my own BirdNET-Go application, https://github.com/tphakala/birdnet-go

Can you be more specific how muting would be implemented in your case? I personally use OBS but I have not yet investigated how it can be controlled from external sources.

**Reply by London-Lass** [reply]
Remote control of OBS is via websocket, there are various implementation of the protocol https://github.com/obsproject/obs-websocket/blob/master/docs/generated/protocol.md

SetInputVolume : Sets the volume setting of an input.

Looks like that would be a straightforward implementation from that side.

**Reply by tphakala** [reply]
Thanks, I need to take a closer look at that.

Regarding implementation for BirdNET-Pi I think it would not be very effective given how BirdNET-Pi currently works, audio is first recorded to disk (10 - 15 seconds based on user configuration) and once recording time limit is hit it will be fed to analyser, so there would be significant delay in mute action.

On BirdNET-Go delay to analyse is 1.5 seconds + BirdNET inference time which is anything from 10ms to 500ms based on hardware it is run on.
